src/Avalonia.Controls/Application.cs

```diff
@@ -190,108 +194,163 @@ public IResourceDictionary Resources
         public WindowCollection Windows { get; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether this instance is existing.
+        /// Gets or sets a value indicating whether this instance is shutting down.
         /// </summary>
         /// <value>
-        ///   <c>true</c> if this instance is existing; otherwise, <c>false</c>.
+        ///   <c>true</c> if this instance is shutting down; otherwise, <c>false</c>.
         /// </value>
-        internal bool IsExiting { get; set; }
+        internal bool IsShuttingDown { get; private set; }
 
         /// <summary>
         /// Initializes the application by loading XAML etc.
         /// </summary>
-        public virtual void Initialize()
+        public virtual void Initialize() { }
+
+        /// <summary>
+        /// Runs the application's main loop.
+        /// This will return when the <see cref="Avalonia.ShutdownMode"/> condition is met
```

Comment:
I think if the summary goes into multiple clauses with descriptions like this, it's probably best to move it to `<remarks>`. So the summary would just be "Run's the application's main loop" and the remarks would explain how the main loop is exited. This makes sense to me because how the application _exits_ isn't really relevant to the summary of the `Run` method.